### ISSUE TITLE
only-arrow-functions: false positive when a function is defined in an ES6 module

### ISSUE THREAD

**Issue by slawomir** [palantir/tslint#1655]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.0.3`
- __Running TSLint via__: grunt-tslint

#### TypeScript code being linted

_// module.ts_
```
// ES6 module
export function foo(): void {
    return undefined;
}
```
_// main.ts_
```
// ES import
import * as Module from 'Module';
```

#### Actual behavior

Error: `non-arrow functions are forbidden`

#### Expected behavior

It should be allowed, since it's a way for declaring modules in ES6

Reference: https://palantir.github.io/tslint/rules/no-namespace/

**Comment by chrismbarr** [comment]
Why not use this?
```ts
// ES6 module
export const foo = (): void => {
    return undefined;
}
```

**Comment by slawomir** [comment]
I've just discovered that ```allow-declarations``` parameter should be used in that scenario.
```"only-arrow-functions": [true, "allow-declarations"]```
If enabled, it's possible to declare es6 module functions, what solves my problem.

**Comment by nchen63** [comment]
Appears to be a non-issue. You can use an arrow function or use the "allow-declarations" option
